cmake/dependencies.cmake

```diff
@@ -114,12 +114,33 @@ endif()
 # GPU Support
 
 if(MICM_ENABLE_CUDA)
-  find_package(CUDA REQUIRED)
-  enable_language(CUDA)
-endif()
+  if(NOT MICM_GPU_TYPE)
+    message(FATAL_ERROR "MICM_GPU_TYPE is not set or is empty. Please provide a GPU type.")
+  endif()
+
+  set(cuda_arch_map_a100 80)
+  set(cuda_arch_map_v100 70)
+  set(cuda_arch_map_h100 90a)
+  set(cuda_arch_map_h200 90a)
+  set(cuda_arch_map_b100 95)
+  set(cuda_arch_map_b200 95)
+  set(MICM_GPU_ARCH ${cuda_arch_map_${MICM_GPU_TYPE}})
+
+  if(NOT MICM_GPU_ARCH)
+    message(FATAL_ERROR "MICM_GPU_TYPE (${MICM_GPU_TYPE}) is not recognized. Available options are [a100, v100, h1/200].")
+  endif()
```

Comment:
Maybe we can combine these output to line 118 and just ask the users to provide a valid GPU type.

Reply:
also is the `MICM_GPU_TYPE` input case sensitive?

Reply:
Based on conversations in the virtual office, we do want to combine these so that is incoming!

Reply:
Removed!